src/widgets/main_widget.py

```diff
@@ -0,0 +1,34 @@
+from PyQt5.QtWidgets import QWidget
+
+from widgets import UNFOLD_BUTTON_WIDTH, UNFOLD_WIDGET_WIDTH
+from widgets.algorithm_widget import AlgorithmWidget
+from widgets.import_widget import ImportWidget
+from widgets.preprocessing_widget import PreprocessingWidget
+from widgets.visualization_widget import VisualizationWidget
```

Comment:
It is good idea to split each class into files.
I think we can simplify imports from widgets package by adding imports in file [\_\_init\_\_.py](src/widgets/__init__.py).
```suggestion
from widgets import AlgorithmWidget, ImportWidget, PreprocessingWidget, VisualizationWidget
```
My suggestion is to change [\_\_init\_\_.py](src/widgets/__init__.py) file. I write my proposition of code in next comment.